include/extractor/edge_based_graph_factory.hpp

```diff
@@ -93,7 +93,8 @@ class EdgeBasedGraphFactory
 
     // The following get access functions destroy the content in the factory
     void GetEdgeBasedEdges(util::DeallocatingVector<EdgeBasedEdge> &edges);
-    void GetEdgeBasedNodes(std::vector<EdgeBasedNode> &nodes);
+    void GetEdgeBasedNodes(EdgeBasedNodeDataContainer &ebg_node_data_container);
+    void GetNodeBasedEdges(std::vector<EdgeBasedNode> &nodes);
```

Comment:
The naming here is a little bit confusing. This function returns a `EdgeBasedNode` but is named `GetNodeBasedEdge`.  I think we should rename that class, it does not represent a single `EdgeBasedNode` but actually a single segment. Maybe `ExternalSegmentData`?